src/main/java/com/datatorrent/netlet/WriteOnlyLengthPrependerClient.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (C) 2016 DataTorrent, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datatorrent.netlet;
+
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.datatorrent.netlet.util.Slice;
+import com.datatorrent.netlet.util.VarInt;
+
+public class WriteOnlyLengthPrependerClient extends WriteOnlyClient
+{
+  private static final Logger logger = LoggerFactory.getLogger(WriteOnlyLengthPrependerClient.class);
+
+  private boolean newMessage = true;
+
+  public WriteOnlyLengthPrependerClient(final int sendBufferCapacity)
+  {
+    this(64 * 1024, sendBufferCapacity);
+  }
+
+  public WriteOnlyLengthPrependerClient(final int writeBufferCapacity, final int sendBufferCapacity)
+  {
+    super(writeBufferCapacity, sendBufferCapacity);
+  }
+
+  @Override
+  public void write() throws IOException
```

Comment:
Correct. It avoids extra slice allocation every 4096 messages or so. We also don't need to handle cases where there are not enough free slots in the send queue for two slices.